persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java

```diff
@@ -67,1091 +67,933 @@
-                  })
-                  .ignoreElements()
-                  .toCompletionStage(null);
-        }
-
-        void deleteBatch(Iterable<Object> keys) {
-            try {
-                int batchSize = 0;
-                WriteBatch batch = new WriteBatch();
-                for (Object key : keys) {
-                    batch.remove(getHandle(calculateSegment(key)), marshall(key));
-                    batchSize++;
-
-                    if (batchSize == configuration.maxBatchSize()) {
-                        batchSize = 0;
-                        writeBatch(batch);
-                        batch = new WriteBatch();
-                    }
-                }
-
-                if (batchSize != 0)
-                    writeBatch(batch);
-            } catch (Exception e) {
-                throw new PersistenceException(e);
-            }
-        }
-
-        abstract void clear(IntSet segments);
-
-        abstract Publisher<K> publishKeys(IntSet segments, Predicate<? super K> filter);
-
-        abstract Publisher<MarshallableEntry<K, V>> publishEntries(IntSet segments, Predicate<? super K> filter,
-                                                                   boolean fetchValue, boolean fetchMetadata);
+               processor.onComplete();
+            }
+         });
+         return processor;
+      });
+   }
+
+   @Override
+   public CompletionStage<Void> addSegments(IntSet segments) {
+      return handler.addSegments(segments);
+   }
+
+   @Override
+   public CompletionStage<Void> removeSegments(IntSet segments) {
+      return handler.removeSegments(segments);
+   }
+
+   @Override
+   public CompletionStage<Boolean> isAvailable() {
+      return blockingManager.supplyBlocking(() -> getLocation().toFile().exists() && getExpirationLocation().toFile().exists(),
+            "rocksdb-available");
+   }
+
+   private abstract class RocksDBHandler {
+
+      abstract RocksDB open(Path location, DBOptions options) throws RocksDBException;
+
+      abstract void close();
+
+      abstract ColumnFamilyHandle getHandle(int segment);
+
+      abstract ColumnFamilyHandle getHandle(Object key);
+
+      ColumnFamilyDescriptor newDescriptor(byte[] name) {
+         ColumnFamilyOptions columnFamilyOptions;
+         if (columnFamilyProperties != null) {
+            columnFamilyOptions = ColumnFamilyOptions.getColumnFamilyOptionsFromProps(columnFamilyProperties);
+            if (columnFamilyOptions == null) {
+               throw log.rocksDBUnknownPropertiesSupplied(columnFamilyProperties.toString());
+            }
+         } else {
+            columnFamilyOptions = new ColumnFamilyOptions();
+         }
+         return new ColumnFamilyDescriptor(name,
+               columnFamilyOptions.setCompressionType(CompressionType.getCompressionType(configuration.compressionType().toString())));
+      }
+
+      CompletionStage<Boolean> contains(int segment, Object key) {
```

Comment:
Never called.

Reply:
Fixed, moved this to the actual store method for better visibility.